### PR TITLE
fix ios keyboard switch bug when use numberic

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/cocos-engine/issues/15328

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
Please describe more about the modification, for example, why it has issue, and how you fix it.

File: native/cocos/ui/edit-box/EditBox-ios.mm [hunk 1]
```diff
@@ -546,7 +546,9 @@ - (id) createTextField:    (const cc::EditBox::ShowInfo*)showInfo {
     ((UITextField*)[ret inputOnToolbar]).text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
     ((UITextField*)[ret inputOnView]).text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
     setTextFieldReturnType((UITextField*)[ret inputOnToolbar], showInfo->confirmType);
+    setTextFieldReturnType((UITextField*)[ret inputOnView], showInfo->confirmType);
     setTextFieldKeyboardType((UITextField*)[ret inputOnToolbar], showInfo->inputType);
+    setTextFieldKeyboardType((UITextField*)[ret inputOnView], showInfo->inputType);
```
Comment (novain1102) [inline native/cocos/ui/edit-box/EditBox-ios.mm:551]: For current implementation, we divide TextField into two parts, `inputOnToolbar` (represents the part related to the keyboard) and `inputOnView`(represents the part displayed when the edit box is clicked). The cause of the inconsistent display issue is not setting the same keyboard for inputOnView and inputOnToolbar. The consequence is that the keyboard related inputOnView will be displayed first, and then the keyboard related inputOnToolbar will be displayed. When the keyboard settings are the same, there is no sense of dissonance.

**Comment by novain1102** [comment]
> Please describe more about the modification, for example, why it has issue, and how you fix it.

Supplemented.

File: native/cocos/ui/edit-box/EditBox-ios.mm (hunk 1 above)
Comment (mmyduckx) [reply native/cocos/ui/edit-box/EditBox-ios.mm:551]: The consequence is that the keyboard related `inputOnView` will be displayed first, and then the keyboard related `inputOnToolbar` will be displayed. When the keyboard settings are the same, there is no sense of dissonance.

**Review by minggo — APPROVED** [review]

**Pull request merged**